packages/facility-server/app/routes/apiv1/surveyResponse.js

```diff
@@ -81,3 +84,54 @@ surveyResponse.post(
     res.send(responseRecord);
   }),
 );
+
+surveyResponse.put(
+  '/complexChartInstance/:id',
+  asyncHandler(async (req, res) => {
+    const {
+      models,
+      body,
+      params,
+      db,
+    } = req;
+
+    const responseRecord = await models.SurveyResponse.findByPk(params.id);
+    if (!responseRecord) {
+      throw new NotFoundError('Response record not found');
+    }
+
+    req.checkPermission('write', subject('Charting', { id: responseRecord.surveyId }));
+
+    const survey = await responseRecord.getSurvey();
+    if (survey.surveyType !== SURVEY_TYPES.COMPLEX_CHART_CORE) {
+      throw new InvalidOperationError('Cannot edit survey responses');
+    }
+
+    const components = await models.SurveyScreenComponent.getComponentsForSurvey(survey.id);
+    const responseAnswers = await models.SurveyResponseAnswer.findAll({
+      where: { responseId: params.id },
+    });
+
+    await db.transaction(async () => {
+      for (const [dataElementId, value] of Object.entries(body.answers)) {
+        // Ignore null values or components that are not in the survey
```

Comment:
Should we error on components that aren't in the survey? That seems like something has gone quite wrong? I guess maybe they filled it out and it has since been updated to remove a question?